var/spack/repos/builtin/packages/libsonata/package.py

```diff
@@ -0,0 +1,62 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Libsonata(CMakePackage):
+    """
+    `libsonata` provides C++ / Python API for reading SONATA Nodes / Edges
+
+    See also:
+    https://github.com/AllenInstitute/sonata/blob/master/docs/SONATA_DEVELOPER_GUIDE.md
+    """
+    homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/common/libsonata"
+    url      = "ssh://bbpcode.epfl.ch/common/libsonata"
+
+    version('dev', git=url, submodules=True)
+
+    variant('mpi', default=False, description="Enable MPI backend")
+
+    depends_on('cmake@3.0:', type='build')
+    depends_on('hdf5+mpi', when='+mpi')
```

Comment:
Do you use hdf5 directly? isn't it a dependency of highfive only?

Reply:
Ironically, it's actually the other way around now: HighFive is brought as a submodule within `libsonata` repo.
Now it's a good time to remember about the potential issue with HighFive version conflict Juan mentioned once.

Reply:
Why HighFive is a submodule?

Reply:
Depends on how you guys want to package this. Even if repo has submodule, you can remove `submodules=True` argument and expose dependency for highfive. As you mentioned, it's "preferable" to not use submodule to avoid version conflicts.

Reply:
> Why HighFive is a submodule?

The rationale for using submodules was building `libsonata` outside of Nix / Spack package managers; as a "standalone" library:
https://bbpcode.epfl.ch/code/#/c/42532/1/CMake/FindHighFive.cmake

> Even if repo has submodule, you can remove submodules=True argument and expose dependency for highfive.

We are using other external libs; but so far all of them could be found in Spack package list, apparently (in addition to `HighFive`, it's `pybind11` and `fmtlib`).
The only obstacle remaining is thus writing CMakeLists.txt in a way that it supports both submodules (for building outside of Spack); and dependencies handed over from Spack.

Do you have an example package for that so I don't re-invent the bicycle here?

Reply:
First thing that comes to my mind is excluding `add_subdirectory(<submodules>)` command in CMakeLists.txt with a patch in Spack recipe. ^^

Reply:
Some packages just have extra option in CMakeList.txt for `-DBUILD_SUBMODULES=OFF` or `-DBUILD_INTERNAL_LIBS=OFF` and then you conditionally include/exclude.  As all dependencies are added to `CMAKE_PREFIX_PATH`, CMake should find those. I remember Adrien removing it once [here](https://github.com/BlueBrain/MVDTool/commit/78b1e915914efa3be5f0ee7b383b59f2e3401b24).

Reply:
As suggested, I've added EXTLIB_FROM_SUBMODULES to `libsonata` CMakeLists.txt to ignore submodules when building from Spack.